src/ui/public/agg_types/metrics/top.js

```diff
@@ -0,0 +1,68 @@
+import { noop } from 'lodash';
+import MetricAggTypeProvider from 'ui/agg_types/metrics/metric_agg_type';
+import topSortEditor from 'ui/agg_types/controls/top_sort.html';
+
+export default function AggTypeMetricTopProvider(Private) {
+  const MetricAggType = Private(MetricAggTypeProvider);
+
+  return new MetricAggType({
+    name: 'top_hits',
+    title: 'Top',
+    makeLabel: function (aggConfig) {
+      const prefix = aggConfig.params.sortOrder.val === 'desc' ? 'Last' : 'First';
+      return `${prefix} ${aggConfig.params.field.displayName}`;
+    },
+    params: [
+      {
+        name: 'field',
+        filterFieldTypes: function (vis, value) {
+          if (vis.type.name === 'table') {
+            return true;
+          }
+          if (vis.type.name === 'pie') {
+            return value === 'number';
+          }
+          return value === 'number' || value === 'date';
```

Comment:
That doesn't really seem like a useful bar graph to me. I think a data table would be more appropriate. Here's a similar example with different data:

![screen shot 2016-10-11 at 1 22 45 pm](https://cloud.githubusercontent.com/assets/6239176/19280959/dd44a006-8fb5-11e6-9da8-92a67841e5b8.png)

Freedom is good but we want to eliminate options that don't make sense. My guess is that most users don't think of dates as being numerical quantities, so a bar chart doesn't make much sense.